### PR TITLE
feat: add translation feature

### PR DESCRIPTION
**Pull request by gabrielfs7**
https://oat-sa.atlassian.net/browse/ADF-1779
https://oat-sa.atlassian.net/browse/ADF-1780
https://oat-sa.atlassian.net/browse/ADF-1781
https://oat-sa.atlassian.net/browse/ADF-1783
https://oat-sa.atlassian.net/browse/HKD-6

# Goal 

- Add new endpoint to get translation information
- Add new feature flag for translation feature

# PRs
* https://github.com/oat-sa/extension-tao-item/pull/660
* https://github.com/oat-sa/extension-tao-test/pull/476
* https://github.com/oat-sa/extension-tao-itemqti/pull/2571
* https://github.com/oat-sa/extension-tao-testqti/pull/2503

# TODO 

- [ ] Update composer dependencies

### CODE REVIEW

File: models/classes/Translation/Service/ResourceMetadataPopulateService.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    private array $metadata;
+
+    public function __construct(Ontology $ontology)
+    {
+        $this->ontology = $ontology;
+    }
+
+    public function addMetadata(string $resourceType, string $metadataUri): void
+    {
+        $this->metadata[$resourceType] = empty($this->metadata[$resourceType]) ? [] : $this->metadata[$resourceType];
```
Comment (shpran) [inline models/classes/Translation/Service/ResourceMetadataPopulateService.php:42]: Can't we use here the following?
```suggestion
        $this->metadata[$resourceType] ??= [];
```

File: models/classes/Translation/Service/ResourceMetadataPopulateService.php [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        $this->metadata[$resourceType] = array_unique(array_merge($this->metadata[$resourceType], [$metadataUri]));
+    }
+
+    public function populate(AbstractResource $resource, core_kernel_classes_Resource $originResource): void
+    {
+        $valueProperty = $this->ontology->getProperty('http://www.w3.org/1999/02/22-rdf-syntax-ns#value');
```
Comment (shpran) [inline models/classes/Translation/Service/ResourceMetadataPopulateService.php:48]:
```suggestion
        $valueProperty = $this->ontology->getProperty(\oat\generis\model\OntologyRdf::RDF_VALUE);
```

File: models/classes/menu/SectionVisibilityFilter.php [hunk 3]
```diff
@@ -73,26 +70,43 @@ public function isVisible(string $section): bool
             $userSettings->getSetting(
                 UserSettingsInterface::INTERFACE_MODE
             ) === GenerisRdf::PROPERTY_USER_INTERFACE_MODE_SIMPLE
-            && in_array($section, self::SIMPLE_INTERFACE_MODE_HIDDEN_SECTIONS)
+            && in_array($sectionPath, self::SIMPLE_INTERFACE_MODE_HIDDEN_SECTIONS)
```
Comment (shpran) [inline models/classes/menu/SectionVisibilityFilter.php:73]: Is it possible to use strict comparison here?
```suggestion
            && in_array($sectionPath, self::SIMPLE_INTERFACE_MODE_HIDDEN_SECTIONS, true)
```

**Review by shpran — APPROVED** [review]
Good job! 👍🏻

File: models/classes/menu/SectionVisibilityFilter.php (hunk 3 above)
Comment (gabrielfs7) [reply models/classes/menu/SectionVisibilityFilter.php:73]: I will, but this is the previous behavior, nothing changed here

**Pull request merged**